src/js/transient-button.js

```diff
@@ -0,0 +1,77 @@
+import Button from './button.js';
+import Component from './component.js';
+import {merge} from './utils/obj';
+import * as Dom from './utils/dom.js';
+
+// Options type will need to extend button options, because `className`, `clickHandler` may be used
+// `position` is redeundant with `className` but allows position to be set without overriding className or vice versa.
+const defaults = {
+  forceTimeout: 4000,
+  position: 'bottom left'
+};
+
+/**
+ * A floating transient button
+ *
+ * @extends Button
+ */
+class TransientButton extends Button {
+  /**
+   *
+   * @param { import('./player').default } player
+   *
+   * @param {object} options
+   */
+  constructor(player, options) {
+    options = merge(defaults, options);
+    super(player, options);
+    this.controlText(options.controlText);
+    this.hide();
+
+    // When shown, the float button will be visible even if the user is inactive.
+    // Clear this if there is any interaction.
+    player.on(['useractive', 'userinactive'], () => {
+      this.removeClass('force-display');
+    });
+  }
+
+  buildCSSClass() {
+    return `vjs-transient-button ${this.options_.position}`;
```

Comment:
How about prefixing the CSS classes `top`, `right`, `bottom`, `left` with `vjs`? The reasoning is that these terms are so generic that they could potentially create a conflict.

Reply:
Very good point